vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ImageService.java

```diff
@@ -250,4 +283,43 @@ public void removeCategory(IVSImage image, ImageCategory category) {
         image.getCategories().remove(category);
         imageRepo.save((VSImage) image);
     }
+
+    /**
+     * This method is used to search the search string specified in the input
+     * 
+     * @param searchTerm - This is the search string which is being searched.
+     * @return list of images based on the search string.
+     */
+    @Override
+    public List <IVSImage> search(String searchTerm) {
+        String likeSearchTerm = "%" + searchTerm + "%";
+        List <VSImage> results = imageRepo.findByFilenameLikeOrNameLikeOrDescriptionLike(likeSearchTerm, likeSearchTerm,
+            likeSearchTerm);
+        List <IVSImage> imageResults = new ArrayList <> ();
+        results.forEach(r -> imageResults.add(r));
+        return imageResults;
+    }
+
+    /**
+     * Method to return the requested images
+     * 
+     * @param pageNo.    if pageNo<1, 1st page is returned, if pageNo>total
+     *                   pages,last page is returned
+     * @param searchTerm - This is the search string which is being searched.
+     * @param images     - This is the result list of search images.
+     * @return list of images in the requested pageNo and requested order.
+     */
+    @Override
+    public List <IVSImage> getImagesForPagination(int pageNo, List <IVSImage> images, String searchTerm) {
```

Comment:
why does this method get a list of images?

Reply:
fixed